test/coordinator-test.js

```diff
@@ -0,0 +1,43 @@
+import { assert } from 'chai';
+import os from 'os';
+import sinon from 'sinon';
+import { getDefaultCPUs, getWorkerCount } from '../lib/coordinator';
+
+describe('coordinator', () => {
+  it('default method returns correct number of cpus', () => {
+    assert.equal(getDefaultCPUs(5), 4, 'getDefaultCPUs returns n - 1 CPUs');
+
+    assert.throws(getDefaultCPUs, TypeError, 'getDefaultCPUs must accept a positive integer');
```

Comment:
These grouped `assert.throws` tests could be refactored to something like

```
[
    () => { getDefaultCPUs('three'); },
    () => { getDefaultCPUs(0); },
    getDefaultCPUs,
].forEach((tryGetDefault) => {
    assert.throws(tryGetDefault, TypeError, 'getDefaultCPUs must accept a positive integer');
});
```

not sure which is better overall for test readability/clarity, maybe doesn't matter too much